cleaner/custom_pre/settings.php

```diff
@@ -0,0 +1,32 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * @package    cleaner_custom_pre
+ * @copyright  2019 Catalyst IT
+ * @author     Srdjan Janković <srdjan@catalyst.net.nz>
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+defined('MOODLE_INTERNAL') || die;
+
+if (!$ADMIN->fulltree) {
+    return;
+}
+
+$settings->add(new admin_setting_configtext('cleaner_custom_pre/sql',
+            new lang_string('sql', 'cleaner_custom_pre'),
+            new lang_string('sqldesc', 'cleaner_custom_pre'), null, PARAM_TEXT));
```

Comment:
I have a suspicion that PARAM_TEXT validation isn't correct. There is a 'customsql' report which I'd use as best practice to mimic:

They use PARAM_RAW:

https://github.com/moodleou/moodle-report_customsql/blob/master/edit_form.php#L61

Bonus points: validation of the sql so that it uses proper table syntax:

https://github.com/moodleou/moodle-report_customsql/blob/master/edit_form.php#L140-L143

Reply:
PARAM_RAW it is. I may go for the bonus later.